src/app/AppComponent.tsx

```diff
@@ -1,27 +1,38 @@
 import React, { Component } from "react";
 import HomeContainer from "./home/HomeContainer";
 import LoginContainer from "./auth/LoginContainer";
-import { Route, Switch, Redirect } from "react-router-dom";
+import { Route, Switch, Redirect, Router } from "react-router-dom";
 import PrivateRoute from "./auth/PrivateRoute";
 import { connect } from "react-redux";
-import { push } from "connected-react-router";
+import { history } from "../helpers";
+import { commonOperations } from "./common/duck";
 
-const AppComponent = ({ alertMessage, alertType, loggedIn, ...props }) => (
-  <div className="app-container-wrapper">
-    {alertMessage && !loggedIn && (
-      <div className={`alert ${alertType}`}>{alertMessage}</div>
-    )}
-    <Switch>
-      <Route exact path="/login" component={LoginContainer} />
-      <PrivateRoute exact path="/" component={HomeContainer} />
-      {localStorage.getItem("currentUser") ? (
-        <Redirect from="*" to="/" />
-      ) : (
-        <Redirect from="*" to="/login" />
-      )}
-    </Switch>
-  </div>
-);
+class AppComponent extends React.Component<any, any> {
+  constructor(props) {
+    super(props);
+
+    const { clearAlerts } = this.props;
+    history.listen((location, action) => {
```

Comment:
I havent seen any edge cases where we could need redux router. What would be the benefit of adding it over just using react-router for now?

Reply:
Because adding redux router means, that all state of application will be in redux, which means that application can react on events inside of React-Redux structure, so in this case better question is, why to use Redux at all, if you won't use it's benefits?